lhotse/recipes/timit.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Xiaomi Corporation (Author: Mingshuang Luo)
+# Apache 2.0
+
+from collections import defaultdict
+
+import os
+import glob
+import json
+import logging
+import shutil
+import tarfile
+import string
+from pathlib import Path
+from tqdm.auto import tqdm
+from typing import Dict, Optional, Union
+from concurrent.futures.thread import ThreadPoolExecutor
+
+from lhotse import validate_recordings_and_supervisions
+from lhotse.audio import Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike, urlretrieve_progress
+
+
+def download_and_unzip(
+        target_dir: Pathlike = '.',
+        force_download: Optional[bool] = False,
+        base_url: Optional[str] = 'https://data.deepai.org/timit.zip') -> None:
+    """
+    Download and unzip the dataset, supporting both TIMIT
+    :param target_dir: Pathlike, the path of the dir to storage the dataset.
+    :param force_download: Bool, if True, download the zips no matter if the zips exists.
+    :param base_url: str, the url of the TIMIT download for free.
+    """
+    target_dir = Path(target_dir)
+    target_dir.mkdir(parents=True, exist_ok=True)
+
+    tar_name = f'timit.zip'
+    tar_path = target_dir / tar_name
+    if force_download or not tar_path.is_file():
+        urlretrieve_progress(f'{base_url}', filename=tar_path, desc=f'Downloading {tar_name}')
+
+
+def prepare_timit(
+        corpus_dir: Pathlike,
+        splits_dir: Pathlike,
+        output_dir: Optional[Pathlike] = None,
+        num_jobs: int = 1
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f'No such directory: {corpus_dir}'
+
+    splits_dir = Path(splits_dir)
+    assert corpus_dir.is_dir(), f'No such directory: {splits_dir}'
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    manifests = defaultdict(dict)
+    dataset_parts = ['TRAIN', 'DEV', 'TEST']
+
+    punctuation_strings = string.punctuation
+
+    with ThreadPoolExecutor(num_jobs) as ex:
+        for part in dataset_parts:
+          wav_files = []
+          file_name = ''
+
+          if part == 'TRAIN':
+            file_name = os.path.join(splits_dir, 'train_samples.txt') 
```

Comment:
You can replace all os.path.join(a, b) with: a / b (this is possible because we use Path objects)